test/e2e/arktos/multi_tenancy/test_suites/crd/test_system_share_crd.yaml

```diff
@@ -0,0 +1,160 @@
+
+  - Command: ${kubectl} get mustshareknowledges --context ${second_tenant}-admin-context
+    OutputShouldContain:
+    - "archimedes-principle"
+
+  - Command: "${kubectl} get mustshareknowledges --all-namespaces --all-tenants -o json 
+              | jq -r '.items[] | [.metadata.name, .metadata.namespace, .metadata.tenant] | @tsv'" 
+    OutputShouldContain:
+    - "pythagorean-theorem	default	${first_tenant}"
+    - "archimedes-principle	default	${second_tenant}"
+
+########################################################################################
+# Verifying tenant deleter works with system-share CRD
+########################################################################################
+
+  - BeforeTestMessage: Verifying tenant deleter works with system-share CRD ...
+    Command: ${kubectl} delete tenant ${first_tenant} 
+    TimeOut: 60
+    OutputShouldBe: "tenant \"${first_tenant}\" deleted\n"
+
+  - Command: ${kubectl} get tenant ${first_tenant} 
+    ShouldFail: true
+    OutputShouldBe: "Error from server (NotFound): tenants \"${first_tenant}\" not found\n"
+
+  - Command: "${kubectl} get mustshareknowledges --all-namespaces --all-tenants -o json 
+              | jq -r '.items[] | [.metadata.name, .metadata.namespace, .metadata.tenant] | @tsv'" 
+    OutputShouldContain:
+    - "archimedes-principle	default	${second_tenant}"
+    OutputShouldNotContain:
+    - "pythagorean-theorem	default	${first_tenant}"
+
+########################################################################################
+# Verify deleting system-share-CRD in system spaces takes effect in all tenants' spaces
+########################################################################################
+
+  - BeforeTestMessage: Verify deleting system-share-CRD in system spaces takes effect in all tenants' spaces
+    Command: ${kubectl} delete -f ${test_data_dir}/forced_share_crd.yaml --tenant system
+    OutputShouldBe: "customresourcedefinition.apiextensions.k8s.io \"mustshareknowledges.learning.wisdom.com\" deleted\n"
+
+  - Command: ${kubectl} get mustshareknowledges --all-namespaces --all-tenants 
+    ShouldFail: true
+    OutputShouldBe: "Error from server (NotFound): Unable to list \"learning.wisdom.com/v1, Resource=mustshareknowledges\": the server could not find the requested resource (get mustshareknowledges.learning.wisdom.com)\n"
+
+  - Command: ${kubectl} get crds --context ${second_tenant}-admin-context
+    OutputShouldBe: "No resources found.\n"
+
+  - Command: ${kubectl} get mustshareknowledges --context ${second_tenant}-admin-context
+    ShouldFail: true
+    OutputShouldBe: "Error from server (NotFound): Unable to list \"learning.wisdom.com/v1, Resource=mustshareknowledges\": the server could not find the requested resource (get mustshareknowledges.learning.wisdom.com)\n"
+
+########################################################################################
+# TODO: add test about resouce discovery after bug 525 is fixed
```

Comment:
this is subject to personal preference but I would suggest using issues to track TODOs instead of comments in code.

Reply:
I remove it. I think it should be a norm for us to add e2e test cases when we fix a bug and add a new feature. I don't need  to add comments for that bug especially.